src/svm/Executable.zig

```
@@ -0,0 +1,494 @@
+                            .src = .r0,
+                            .off = 0,
+                            .imm = @truncate(@as(u64, @bitCast(operands[1].integer))),
+                        },
+                        .load_reg => .{
+                            .opcode = @enumFromInt(bind.opc),
+                            .dst = operands[0].register,
+                            .src = operands[1].memory.base,
+                            .off = operands[1].memory.offset,
+                            .imm = 0,
+                        },
+                        .store_reg => .{
+                            .opcode = @enumFromInt(bind.opc),
+                            .dst = operands[0].memory.base,
+                            .src = operands[1].register,
+                            .off = operands[0].memory.offset,
+                            .imm = 0,
+                        },
+                        .store_imm => .{
+                            .opcode = @enumFromInt(bind.opc),
+                            .dst = operands[0].memory.base,
+                            .src = .r0,
+                            .off = operands[0].memory.offset,
+                            .imm = @bitCast(@as(i32, @intCast(operands[1].integer))),
+                        },
+                        .endian => |bits| .{
+                            .opcode = @enumFromInt(bind.opc),
+                            .dst = operands[0].register,
+                            .src = .r0,
+                            .off = 0,
+                            .imm = bits,
+                        },
+                        .call_imm => inst: {
+                            const is_label = operands[0] == .label;
+                            if (is_label) {
+                                const label = operands[0].label;
+                                const target_pc = labels.get(label) orelse
+                                    std.debug.panic("label not found: {s}", .{label});
```

feels like this should just return an error?

---

See the above comment about panics: https://github.com/Syndica/sig/pull/478#discussion_r1905849132